libprecompiled/UpdateContractStatusPrecompiled.cpp

```diff
@@ -70,26 +77,161 @@ bool UpdateContractStatusPrecompiled::checkAddress(Address const& contractAddres
+            entry->setField(storagestate::STORAGE_VALUE, "");
+            table->update(storagestate::ACCOUNT_CODE, entry, table->newCondition());
+            entry = table->newEntry();
+            entry->setField(storagestate::STORAGE_VALUE, toHex(EmptySHA3));
+            table->update(storagestate::ACCOUNT_CODE_HASH, entry, table->newCondition());
+            entry = table->newEntry();
+            entry->setField(storagestate::STORAGE_VALUE, "false");
+            result = table->update(storagestate::ACCOUNT_ALIVE, entry, table->newCondition());
+        }
+    }
+
+    getErrorCodeOut(out, result);
+}
+
+void UpdateContractStatusPrecompiled::setFrozenStatus(
+    ExecutiveContext::Ptr context, bytesConstRef data, bytes& out, Address const& origin)
+{
+    (void)origin;
+    dev::eth::ContractABI abi;
+    int result = 0;
+
+    Address contractAddress;
+    bool frozen;
+    abi.abiOut(data, contractAddress, frozen);
+
+    if (!checkAddress(contractAddress))
+    {
+        PRECOMPILED_LOG(ERROR) << LOG_BADGE("UpdateContractStatusPrecompiled")
+                               << LOG_DESC("address invalid") << LOG_KV("address", contractAddress);
+        result = CODE_ADDRESS_INVALID;
+    }
+    else
+    {
+        /// TODO:Permission check
+        std::string tableName = precompiled::getContractTableName(contractAddress);
+        PRECOMPILED_LOG(DEBUG) << LOG_BADGE("UpdateContractStatusPrecompiled")
+                               << LOG_DESC("call setFrozenStatus")
+                               << LOG_KV("contract table name", tableName)
+                               << LOG_KV("frozen status", frozen);
+
+        Table::Ptr table = openTable(context, tableName);
+        if (table)
+        {
+            auto entries = table->select(storagestate::ACCOUNT_ALIVE, table->newCondition());
+            if (entries->size() > 0 && entries->get(0) &&
+                "true" == entries->get(0)->getField(storagestate::STORAGE_VALUE))
+            {  // do only in alive
+                entries = table->select(storagestate::ACCOUNT_FROZEN, table->newCondition());
+                auto entry = table->newEntry();
+                std::string status = frozen ? "true" : "false";
+                entry->setField(storagestate::STORAGE_VALUE, status);
+                if (entries->size() != 0u)
+                {
+                    result =
+                        table->update(storagestate::ACCOUNT_FROZEN, entry, table->newCondition());
+                }
+                else
+                {
+                    result = table->insert(storagestate::ACCOUNT_FROZEN, entry);
+                }
+            }
+            else
+            {
+                result = CODE_INVALID_CONTRACT_KILLED;
+            }
+        }
+    }
+    getErrorCodeOut(out, result);
+}
+
+void UpdateContractStatusPrecompiled::query(
+    ExecutiveContext::Ptr context, bytesConstRef data, bytes& out)
+{
+    dev::eth::ContractABI abi;
+    int result = 0;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'result' is assigned a value that is never used.](https://app.codacy.com/manual/fisco/FISCO-BCOS/pullRequest?prid=4839130)